presto-main/src/main/java/io/prestosql/util/DateTimeUtils.java

```diff
@@ -196,35 +184,42 @@ public static long convertToTimestamp(String value)
         }
     }
 
-    public static long parseTimestampWithTimeZone(String value)
+    /**
+     * Parse a legacy TIMESTAMP type. The string is interpreted in {@code timeZoneKey} zone.
+     */
+    @Deprecated
+    public static long parseLegacyTimestamp(TimeZoneKey timeZoneKey, String value)
     {
-        DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.withOffsetParsed().parseDateTime(value);
-        return packDateTimeWithZone(dateTime);
+        return LEGACY_TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
     }
 
     /**
-     * Parse a string (optionally containing a zone) as a value of TIMESTAMP WITH TIME ZONE type.
+     * Parse a string (optionally containing a zone) as a value of TIMESTAMP type using legacy semantics.
      * If the string doesn't specify a zone, it is interpreted in {@code timeZoneKey} zone.
-     * <p>
-     * For example: {@code "2000-01-01 01:23:00"} is parsed to TIMESTAMP WITH TIME ZONE
-     * {@code 2000-01-01T01:23:00 <provided zone>} and {@code "2000-01-01 01:23:00 +01:23"}
-     * is parsed to TIMESTAMP WITH TIME ZONE {@code 2000-01-01T01:23:00.000+01:23}.
-     *
-     * @return stack representation of TIMESTAMP WITH TIME ZONE type
+     * If the timestamp is in a gap in the provided timezone, the method throws an exception.
      */
-    public static long convertToTimestampWithTimeZone(TimeZoneKey timeZoneKey, String timestampWithTimeZone)
+    @Deprecated
+    public static long convertToLegacyTimestamp(TimeZoneKey timeZoneKey, String value)
     {
-        DateTime dateTime = TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).withOffsetParsed().parseDateTime(timestampWithTimeZone);
-        return packDateTimeWithZone(dateTime);
+        return TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
     }
 
     /**
-     * Parse a string (optionally containing a zone) as a value of TIMESTAMP type.
-     * If the string doesn't specify a zone, it is interpreted in {@code timeZoneKey} zone.
+     * Parse a string as a value of TIMESTAMP type.
      *
-     * @return stack representation of legacy TIMESTAMP type
-     * @deprecated applicable in legacy timestamp semantics only
+     * @return stack representation of TIMESTAMP type
      */
+    public static long parseTimestamp(String value)
```

Comment:
I assume the changes in this class in `fixup: Clarify parsing of timestamps and time types` are just reorder.

Reply:
Yes, it's easier to see once I squash the fixup commit.

Reply:
to be clear: I am not questioning this. just letting know i passed over this "without deep read"

Reply:
Yup, got it. Take a look now. I squashed the fixups, so now it should be clear what the actual changes are in that original commit.